### PR TITLE
UIDATIMP-709: Cover `<ListView>` component with tests

### PR DESCRIPTION
**Pull request by OleksandrHladchenko1**
## Purpose
Cover ListView component with tests

## Approach
Covered ListView component with tests using RTL + Jest

## Refs
https://issues.folio.org/browse/UIDATIMP-709

## Screenshots

image

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +28,11 @@
 * When closing the log summary, user sometimes goes to unexpected place (UIDATIMP-949)
 * After editing the match profile the data does not change (UIDATIMP-951)
 
+
```
Comment (IvanKryzhanovskyi) [inline CHANGELOG.md:31]: remove this extra space, please

**Review by IvanKryzhanovskyi — APPROVED** [review]

File: src/components/ListView/ListView.test.js [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+        parentProfiles: 'testParent',
+        childProfiles: 'testChild',
+      },
+      hasLoaded: true,
+    },
+    setList: jest.fn(),
+    RecordView: jest.fn(),
+    history: { push: history.push },
+  };
+};
```
Comment (mariia-aloshyna) [inline src/components/ListView/ListView.test.js:71]:
```suggestion
const listViewProps = listViewFilterProps => ({
    ...listViewFilterProps,
    selectedRecord: {
      record: {
        id: 'testId1',
        parentProfiles: 'testParent',
        childProfiles: 'testChild',
      },
      hasLoaded: true,
    },
    setList: jest.fn(),
    RecordView: jest.fn(),
    history: { push: history.push },
});
```

File: src/components/ListView/ListView.test.js [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+
+const listViewPropsActionProfiles = {
+  resources: {
```
Comment (mariia-aloshyna) [inline src/components/ListView/ListView.test.js:74]: Use `buildResources` and `buildMutator` helpers from `@folio/stripes-data-transfer-components/test/helpers`

File: src/components/ListView/ListView.test.js [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+  renderHeaders: () => {
+    return {
+      name: 'Name',
+      action: 'Action',
+      tags: 'Tags',
+      updated: 'Updated',
+      updatedBy: 'Updated By',
+    };
+  },
+};
```
Comment (mariia-aloshyna) [inline src/components/ListView/ListView.test.js:157]:
```suggestion
  renderHeaders: () => ({
      name: 'Name',
      action: 'Action',
      tags: 'Tags',
      updated: 'Updated',
      updatedBy: 'Updated By',
  }),
};
```

File: src/components/ListView/ListView.test.js [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+
+const listViewPropsFileExtensions = {
```
Comment (mariia-aloshyna) [inline src/components/ListView/ListView.test.js:159]: Same here

File: src/components/ListView/ListView.test.js [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+describe('ListView', () => {
+  afterEach(() => {
+    history.push.mockClear();
+  });
+
+  it('should be rendered as File Extensions', () => {
+    const { getByText } = renderListView(listViewProps(listViewPropsFileExtensions));
+
+    expect(getByText('File Extensions Label')).toBeDefined();
+  });
```
Comment (mariia-aloshyna) [inline src/components/ListView/ListView.test.js:262]:
```suggestion
describe('when profile type is File extention', () => {
  it('should render correct label');
});
```

File: src/components/ListView/ListView.test.js [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+
+  it('should be rendered as Action Profiles', () => {
+    const { getByText } = renderListView(listViewProps(listViewPropsActionProfiles));
+
+    expect(getByText('Action Profiles Label')).toBeDefined();
+  });
```
Comment (mariia-aloshyna) [inline src/components/ListView/ListView.test.js:268]: Same here

File: src/components/ListView/ListView.test.js [hunk 8]
```diff
@@ -0,0 +1,321 @@
...
+
+  describe('when click on show Restore Modal', () => {
```
Comment (mariia-aloshyna) [inline src/components/ListView/ListView.test.js:270]:
```suggestion
  describe('when clicking on show Restore Modal', () => {
```

File: src/components/ListView/ListView.test.js [hunk 9]
```diff
@@ -0,0 +1,321 @@
...
+    describe('when click on Actions button', () => {
```
Comment (mariia-aloshyna) [inline src/components/ListView/ListView.test.js:271]:
```suggestion
    describe('when clicking on Actions button', () => {
```

File: src/components/ListView/ListView.test.js [hunk 10]
```diff
@@ -0,0 +1,321 @@
...
+      it('dropdown should be shown', () => {
+        const { getByText } = renderListView(listViewProps(listViewPropsFileExtensions));
+        const actionsButton = getByText('Actions');
+
+        const resetAll = getByText('Reset all extension mappings to system defaults');
+
```
Comment (mariia-aloshyna) [inline src/components/ListView/ListView.test.js:277]:
```suggestion
        const resetAllItem = getByText('Reset all extension mappings to system defaults');
        const newFileExtensionsItem = getByText('New file extensions');
```

File: src/components/ListView/ListView.test.js [hunk 11]
```diff
@@ -0,0 +1,321 @@
...
+        fireEvent.click(actionsButton);
+
+        expect(resetAll).toBeDefined();
+      });
+    });
+
+    describe('when click on Reset All File Extensions', () => {
+      it('modal should be shown', () => {
```
Comment (mariia-aloshyna) [inline src/components/ListView/ListView.test.js:285]:
```suggestion
      it('modal window should be shown', () => {
```

File: src/components/ListView/ListView.test.js [hunk 12]
```diff
@@ -0,0 +1,321 @@
...
+        const resetAll = getByText('Reset all extension mappings to system defaults');
+
+        fireEvent.click(resetAll);
+
+        const modalContent = getByText('Confirmation modal');
+
+        expect(modalContent).toBeDefined();
+      });
+
+      describe('when modal is opened', () => {
```
Comment (mariia-aloshyna) [inline src/components/ListView/ListView.test.js:300]:
```suggestion
      describe('when clicking on modal close button', () => {
```

File: src/components/ListView/ListView.test.js [hunk 13]
```diff
@@ -0,0 +1,321 @@
...
+        it('should close modal', async () => {
```
Comment (mariia-aloshyna) [inline src/components/ListView/ListView.test.js:301]:
```suggestion
        it('modal window should be closed', async () => {
```

**Review by mariia-aloshyna — APPROVED** [review]

**Pull request merged**